slack_bolt/request/async_internals.py

```diff
@@ -41,6 +44,15 @@ def build_async_context(
     channel_id = extract_channel_id(body)
     if channel_id:
         context["channel_id"] = channel_id
+    function_execution_id = extract_function_execution_id(body)
+    if function_execution_id:
+        context["function_execution_id"] = function_execution_id
+        function_bot_access_token = extract_function_bot_access_token(body)
+        if function_bot_access_token is not None:
```

Comment:
Perhaps we should throw an exception if we get here and there is no bot access token?

Reply:
We do not need to through an exception here 🤔  this logic extracts the `function_bot_access_token` if it is found in the payload, if it isn't then the payload is not a custom step related requests so nothing happens